zcash_client_sqlite/src/wallet/transparent.rs

```diff
@@ -1126,34 +1164,49 @@ pub(crate) fn transaction_data_requests<P: consensus::Parameters>(
     let chain_tip_height =
         super::chain_tip_height(conn)?.ok_or(SqliteClientError::ChainHeightUnknown)?;
 
+    debug!(
+        "Generating transaction data requests as of chain tip height {}",
+        chain_tip_height
+    );
+
     // We cannot construct address-based transaction data requests for the case where we cannot
     // determine the height at which to begin, so we require that either the target height or mined
     // height be set.
     let mut spend_requests_stmt = conn.prepare_cached(
         "SELECT
             ssq.address,
-            IFNULL(t.target_height, t.mined_height)
+            COALESCE(tro.max_observed_unspent_height, t.mined_height, t.target_height)
```

Comment:
Why do we now prefer `t.mined_height` over `t.target_height`? Is this fixing a bug, or is this a semantic change? (The documentation above once updated might help explain.)

Reply:
It now occurs to me that there was already a subtle bug here and it's unrelated to this change; the existing code incorrectly made the assumption that if a transaction was mined in a block, that its outputs were unspent as of the end of that block. However, that's not the case for transparent outputs, which can be spent with zero confirmations.

Reply:
I believe the correct behavior here is:

* Change the semantics of  `max_observed_unspent_height` (which is not read prior to this PR) to have the meaning "the UTXO is known to be unspent as of the *end* of the given block"

and then use:

```
COALESCE(tro.max_observed_unspent_height + 1, t.mined_height, t.target_height)
```